### ISSUE TITLE
Include user comments and screenshots in Azure DevOps Issues created from Sentry User Feedback

### ISSUE THREAD

**Issue by kerenkhatiwada** [getsentry/sentry#75775]
### Problem Statement

When creating a new issue from within Sentry's User Feedback to a connected Azure DevOps instance, the new Azure DevOps item does not include the comment the user sent into the description box, just a link back to the event in Sentry. While it's possible to manually add the user comment when creating the Azure DevOps issues manually, it would be great to have the Azure DevOps issue include the user feedback's comment and include screenshots by default.

### Product Area

User Feedback

**Comment by bruno-garcia** [comment]
Do you create the issue manually?

If so, a template shows up like this:


![Image](https://github.com/user-attachments/assets/2f9e9707-b8bd-4b72-9f6a-da4d19eb4f14)


Do you see the message there? Perhaps under `<!--`?

**Comment by johnwc** [comment]
@bruno-garcia You are referring to a sentry issue, we are referring to a User Feedback where a user has entered a comment and potentially added a screenshot from the browser.

**Comment by bruno-garcia** [comment]
Right, but since we use the same component, and with other providers, I was asking if you see the comment in it.

I have a GitHub integration with my project and I see the comment, eg:


![Image](https://github.com/user-attachments/assets/c36ca0a9-2909-4910-89fa-3778188c4417)

**Comment by johnwc** [comment]
I created the request because we see nothing in the dialog's Description you are screenshotting. Nor do we see any screenshots added to the issue in Azure DevOps. We also do not see any of those extra options like Assignee.

![Image](https://github.com/user-attachments/assets/4a7c8030-e0df-4695-b444-a993f9d45bf1)

**Comment by michellewzhang** [comment]
@bruno-garcia @johnwc 's screenshot is accurate -- we use a different template for non-github/jira linked issues/feedback. the problem is that this template is not azure devops specific -- so changing the template would affect all other integrations sentry offers, which could be problematic. we were able to make this change easily for github/jira linked tickets because they have specific templates, so there wasn't a huge risk in updating those default descriptions. 

@johnwc we don't currently support having screenshots in any of the linked integration forms unfortunately, this is probably not something we can support anytime soon due to privacy concerns.

**Comment by bruno-garcia** [comment]
> we use a different template for non-github/jira linked issues/feedback. the problem is that this template is not azure devops specific

Could we create a new, azure specific template based on the current one. And modify it only for user-feedback so that we add the comment in it?

**Comment by michellewzhang** [comment]
> > we use a different template for non-github/jira linked issues/feedback. the problem is that this template is not azure devops specific
> 
> Could we create a new, azure specific template based on the current one. And modify it only for user-feedback so that we add the comment in it?

this is hard, it's something that the integrations team would have to get involved with. it also seems like pretty old code (github one has code from 3 years ago)

**Comment by johnwc** [comment]
@michellewzhang I'm sorry but none of this is making any logical sense to me.

> we don't currently support having screenshots in any of the linked integration forms unfortunately, this is probably not something we can support anytime soon due to privacy concerns.

If a customer goes out of their way to screenshot a page and report a bug, how is affected with privacy concerns? The user purposely created the screenshot. How is it a privacy concern if it is in our private devops, but not a privacy concern if it in sentry? 

> we were able to make this change easily for github/jira linked tickets because they have specific templates, so there wasn't a huge risk in updating those default descriptions.

All Azure DevOps items from Epics to Bugs support attaching files, so how is it different from the way you are creating specific template for github/jira? If you are not referring to attaching files(screenshots), devops supports the same markdown in the description as github/jira.

> i'm also not sure making a specific template for every single integration is the right way to go -- sentry has so many integrations, so not sure if we want to make one for every single one

Why would it not be beneficial in all integrations to have the user's actual feedback in the comment of the item being created? Just like you do for github/jira.

**Comment by michellewzhang** [comment]
@johnwc sorry for the confusion -- hope this message can clarify some things. so we don't support automatic adding of screenshots for github or jira integrations either; it's something we'd have to discuss/plan first before implementing. the screenshots are also not readily available on the issue object itself; we'd have to call a separate endpoint to get the attachments so it's not a trivial fix. we might have a ticket somewhere for adding screenshots to the default message already (cc @jas-kas).

to your point about creating a new integration template for every integration we have: our specific templates for github/jira are not super straightforward -- we're not the team who wrote them, so i'd need to discuss with our integrations team to get more insights into how to create new templates for the remaining ones. it's complex so takes a lot of time and not sure if any of our current teams have bandwidth to take this on currently.

for now, i can put up a PR for our generic linked integration template that adds the feedback message and other evidence in the default template (but it won't include the screenshots yet).

**Comment by michellewzhang** [comment]
PR: https://github.com/getsentry/sentry/pull/76016

**Comment by jas-kas** [comment]
Feature request ticket for adding screenshots to external tickets: https://github.com/getsentry/sentry/issues/76063

**Comment by johnwc** [comment]
@michellewzhang can we also get the link to the replay session added to the description content? (If there is a replay for the user feedback/issue in sentry)

**Comment by michellewzhang** [comment]
hi @johnwc, thanks for reaching out -- we currently have a feature request for this on our backlog: https://github.com/getsentry/sentry/issues/61768. it's on our radar!

**Comment by michellewzhang** [comment]
@johnwc changes have been deployed & you should be able to see the user message by default in the create ticket description now! let us know if you run into issues

**Comment by johnwc** [comment]
@michellewzhang I see it coming through, but the comment it creates is not marked as being markdown content. So it does not format correctly. I have to edit the comment and click that it is markdown.

![Image](https://github.com/user-attachments/assets/07166928-e304-45e9-8636-c57d75cff410)

**Comment by michellewzhang** [comment]
> @michellewzhang I see it coming through, but the comment it creates is not marked as being markdown content. So it does not format correctly. I have to edit the comment and click that it is markdown.
> 
> ![Image](https://github.com/user-attachments/assets/07166928-e304-45e9-8636-c57d75cff410)
> 
> And this is just a preference, but could the link to sentry be placed after the description? The description is the most important thing, so it should be the first thing.

hi @johnwc, can you send an example of what the correct formatting would be?

**Comment by johnwc** [comment]
@michellewzhang well the comment's `format` seems to be set to html, and it should be set to markdown. 
https://learn.microsoft.com/en-us/rest/api/azure/devops/wit/comments/add-work-item-comment?view=azure-devops-rest-7.1&tabs=HTTP#commentformat

**Comment by bruno-garcia** [comment]
Looking at the PR: https://github.com/getsentry/sentry/pull/76016/files
We're only adding markdown, and no reference of explicitly setting the type to `html`.

Maybe azuredevops defaults to `html`, and we need to explicitly set something to have it realize it's markdown

**Comment by johnwc** [comment]
> Looking at the PR: https://github.com/getsentry/sentry/pull/76016/files
> We're only adding markdown, and no reference of explicitly setting the type to `html`.
> 
> Maybe azuredevops defaults to `html`, and we need to explicitly set something to have it realize it's markdown (there's no mention of default on their docs).
> 
> We might be able to do that with a `?format=markdown` : https://learn.microsoft.com/en-us/rest/api/azure/devops/wit/comments/add-work-item-comment?view=azure-devops-rest-7.1&tabs=HTTP#uri-parameters

I do believe you are correct as far as html being default setting, as markdown was just recently added. Prior we could only use html.
